torch/utils/cpp_extension.py

```diff
@@ -275,14 +322,21 @@ def unix_wrap_compile(obj, src, ext, cc_args, extra_postargs, pp_opts):
             try:
                 original_compiler = self.compiler.compiler_so
                 if _is_cuda_file(src):
-                    nvcc = _join_cuda_home('bin', 'nvcc')
+                    nvcc = (_join_rocm_home('bin', 'hipcc') if ROCM_HOME else _join_cuda_home('bin', 'nvcc'))
                     if not isinstance(nvcc, list):
                         nvcc = [nvcc]
                     self.compiler.set_executable('compiler_so', nvcc)
                     if isinstance(cflags, dict):
                         cflags = cflags['nvcc']
-                    cflags = COMMON_NVCC_FLAGS + ['--compiler-options',
+                    if ROCM_HOME:
+                        cflags = COMMON_HIPCC_FLAGS + cflags + _get_rocm_arch_flags(cflags)
+                    else:
+                        cflags = COMMON_NVCC_FLAGS + ['--compiler-options',
                                                   "'-fPIC'"] + cflags + _get_cuda_arch_flags(cflags)
+                elif ROCM_HOME:
```

Comment:
If possible we should enable some tests in `test_cpp_extension` to run in our ROCM CI (I assume they don't run right now).

Reply:
@zou3519 - I just removed `test_cpp_extensions_aot_no_ninja` from `ROCM_BLACKLIST`. However, the CI still skipped it. Is there anywhere else as well I should change?

Reply:
I don't know how the mechanism works. @iotamudelta do you know how we can trigger the rocm tests on this PR?

Reply:
This is a deficiency (well, probably on purpose, but not good here) in the definition exclude_test that causes prefixes in the blacklist to match:
https://github.com/pytorch/pytorch/blob/44af8ee6cd3073ef2c6c6e8a61cfdde5dff9b345/test/run_test.py#L404
A quick fix could be to allow the blacklist to move to either move to regular expressions or allow "foo$" manually to mean "=='foo'" instead of "startswith(foo)".
Or one could rename the tests, no_ninja to the normal one and suffix the ninja one...

Reply:
@t-vi, That would be a change with implications throughout the pytorch test suite. Would you recommend not touching it for this pull request and look at it separately?

Reply:
Upon further inspection, I'm not so sure whether the prefix matching even is the intended behaviour (e.g. non-x64 windows blacklists both cpp_extension_aot and cpp_extension_aot_no_ninja).
I think that:
- we really do want the test,
- renaming the ninja-enabled test to _ninja  that is the prefix is probably best and I would say it's legitimate to do it in this PR. Changing the overall behaviour less so.

Reply:
Note that renaming the test also needs updating the windows blacklist etc.

Reply:
Oh, so that's the problem. I agree with @t-vi that renaming the ninja-enabled test is the easiest thing to do for now. Long-term we should file an issue and figure out if the prefix matching behavior is intended behavior or not; afaict it isn't documented in the test runner.

Reply:
Thank you for the direction. Renamed the tests, the CI run should now have cpp_extensions test enabled for ROCm